src/core/net/dnssd_server.hpp

```diff
@@ -232,46 +236,111 @@ class Server : public InstanceLocator, private NonCopyable
+    void                    HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessageInfo);
+    void                    ProcessQuery(const Header &          aRequestHeader,
+                                         Message &               aRequestMessage,
+                                         Message &               aResponseMessage,
+                                         const Ip6::MessageInfo &aMessageInfo);
+    static Header::Response AddQuestions(const Header &    aRequestHeader,
+                                         const Message &   aRequestMessage,
+                                         Header &          aResponseHeader,
+                                         Message &         aResponse,
+                                         uint8_t &         aResolveAdditional,
+                                         NameCompressInfo &aCompressInfo);
+    static otError          AppendQuestion(const char *      aName,
+                                           const Question &  aQuestion,
+                                           Message &         aMessage,
+                                           NameCompressInfo &aCompressInfo);
+    static otError          AppendPtrRecord(Message &         aMessage,
+                                            const char *      aServiceName,
+                                            const char *      aInstanceName,
+                                            uint32_t          aTtl,
+                                            NameCompressInfo &aCompressInfo);
+    static otError          AppendSrvRecord(Message &         aMessage,
+                                            const char *      aInstanceName,
+                                            const char *      aHostName,
+                                            uint32_t          aTtl,
+                                            uint16_t          aPriority,
+                                            uint16_t          aWeight,
+                                            uint16_t          aPort,
+                                            NameCompressInfo &aCompressInfo);
+    static otError          AppendTxtRecord(Message &         aMessage,
+                                            const char *      aInstanceName,
+                                            const void *      aTxtData,
+                                            uint16_t          aTxtLength,
+                                            uint32_t          aTtl,
+                                            NameCompressInfo &aCompressInfo);
+    static otError          AppendAaaaRecord(Message &           aMessage,
+                                             const char *        aHostName,
+                                             const Ip6::Address &aAddress,
+                                             uint32_t            aTtl,
+                                             NameCompressInfo &  aCompressInfo);
```

Comment:
It seems better to have those static methods as member methods of the `DnsQueryTransaction` class. Thoughts?

Reply:
These methods are also used when resolving by SRP. They are not specific to Discovery Proxy.

Reply:
Let's postpone this major refactoring and deliver the working feature first.